sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -354,7 +358,10 @@ inline __SYCL_ALWAYS_INLINE float round_to_tf32(float &a) {
   uint32_t tmp_uint = reinterpret_cast<uint32_t &>(a);
   tmp_uint += 0x1000u;
   tmp_uint &= 0xFFFFE000u;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
```

Comment:
Why we need to disable this diagnostic? Also what happens if we compile the code via MSVS or clang?

Reply:
line363 will break strict-aliasing assumption when compileing level_zeor.cpp.
i didn't figure how to fix it so i did a workaround. do you have any idea for this?

Reply:
I would try `restrict` keyword, but I don't understand why this warning showed up during this patch implementation.

Reply:
Like here: https://godbolt.org/z/nMzaa1dqz but I don't see any warnings in the first place.

Reply:
> Like here: https://godbolt.org/z/nMzaa1dqz but I don't see any warnings in the first place.

you forgot to add "-Werror=strict-aliasing -O3"

Reply:
Yeap, like this it shows up, thanks! Anyway it's more of a question, then a request of changes. The last word is on the codeowners side.

Reply:
@MrSidims did you figure out why "float *ret = reinterpret_cast<float *>(&tmp_uint)" won't break strict-aliasing? i thought ret is a different type pointer and it might break strict-aliasing.

Reply:
I feel like it just helps with avoiding diagnostics from clang and not with resolving aliasing issues. [The later commit ](https://github.com/intel/llvm/pull/7894/commits/e1adf3edf47370ccdd4ae6532b7d43832a31230d) fixes it.

Reply:
you mean there was alias issue in "float *ret = reinterpret_cast<float *>(&tmp_uint)", but compiler didn't report it?